tensor_comprehensions/README.md

```diff
@@ -0,0 +1,63 @@
+
+# Using Tensor Comprehensions with PyTorch
+
+A blogpost on Tensor Comprehensions can be read [here](https://research.fb.com/announcing-tensor-comprehensions/).
+
+We provide integration of Tensor Comprehensions (TC) with PyTorch for both training
+and inference purposes. Using TC, you can express an operator using [Einstein
+notation](https://obilaniu6266h16.wordpress.com/2016/02/04/einstein-summation-in-numpy/)
+and get the fast CUDA code for that layer with a few lines of code. By providing
+TC integration with PyTorch, we hope to make it further easy to write new
+operations with TC.
+
+Here is what the PyTorch-TC package provides:
+
+- inputs and outputs to functions are `torch.*Tensor`s
+- Integration with PyTorch `autograd`: if you specify forward and backward functions, you get an autograd function that takes `Variable` as input and returns `Variable` as output.
+- autotuner results can be cached to a file (for reuse)
+
+
+## Installation
+
+To make it easy to use TC, we provide conda packages for it. Follow the instructions at
+our documentation [here](https://facebookresearch.github.io/TensorComprehensions/framework/pytorch_integration/getting_started.html#installation) for installing the PyTorch TC package.
+
+## Examples and Documentation
+
+In order to explore Tensor Comprehensions (TC), there are few helpful resources to get started:
+
+1. We provide **examples** of TC definitions covering wide range of Deep Learning layers. Please look at `test_python/layers/` for various layers definitions.
+
+2. [TC Documentation](https://facebookresearch.github.io/TensorComprehensions/index.html)
+is a very helpful resource to understand how Tensor Comprehensions are expressed. The sections on
+[introduction](https://facebookresearch.github.io/TensorComprehensions/introduction.html),
+[range inference](https://facebookresearch.github.io/TensorComprehensions/inference.html),
+[semantics](https://facebookresearch.github.io/TensorComprehensions/semantics.html)
+are particularly helpful to get insights into writing Tensor Comprehensions.
+
+3. **Autotuner**: TC provides an evolutionary search based algorithm to automatically tune the kernel.
+You can read briefly about autotuner [here](https://facebookresearch.github.io/TensorComprehensions/framework/pytorch_integration/autotuning_layers.html) and look at various tests at `test_python/layers/test_autotuner.py`.
+
+4. To construct a TC autograd function, `test_python/layers/test_convolution_train-py` is one self-descriptive example.
+
+### A simple Example
+
+```python
+import tensor_comprehensions as tc
+import torch
+lang = """
+def matmul(float(M,N) A, float(N,K) B) -> (output) {
+  output(i, j) +=! A(i, kk) * B(kk, j)
+}
+"""
+matmul = tc.define(lang, name="matmul")
+mat1, mat2 = torch.randn(3, 4).cuda(), torch.randn(4, 5).cuda()
+out = matmul(mat1, mat2)
+```
+
+## Communication
+
+* **Email**: prigoyal@fb.com
```

Comment:
```tensorcomp@fb.com```

Reply:
this section is just not needed, I'll remove it, the communication will go on the README page